weldx/transformations/cs_manager.py

```diff
@@ -1387,12 +1394,14 @@ def get_cs(
 
             if lcs.is_time_dependent:
                 if not lcs.has_reference_time and self.has_reference_time:
-                    time_lcs = time_interp + (time_ref_interp - self.reference_time)
-                    lcs = lcs.interp_time(time_lcs)
-                    lcs.reset_reference_time(self.reference_time)
-                    lcs.reset_reference_time(time_ref_interp)
+                    lcs = lcs.interp_time(time - self.reference_time)
+                    lcs = LocalCoordinateSystem(
+                        lcs.orientation.data, lcs.coordinates.data, time
+                    )
                 else:
-                    lcs = lcs.interp_time(time_interp, time_ref_interp)
+                    if time_ref is None and len(path_edges) == 1:
+                        time_ref = lcs.reference_time
+                    lcs = lcs.interp_time(time, time_ref)
```

Comment:
The missing branch was actually an untested bug in the previous implementation. Maybe it does make sense to handle the special case that we have just a single edge outside the loop. Should yield better performance and also makes the code less hard to understand.

Reply:
Done in PR #468